test-suite/crosscurrencyratehelpers.cpp

```diff
@@ -0,0 +1,276 @@
+
+            return instruments;
+        }
+
+        std::vector<ext::shared_ptr<Swap> >
+        buildXccyBasisSwap(const Date& start,
+                           const XccyTestDatum& q,
+                           Real fxSpot,
+                           bool isFxBaseCurrencyCollateralCurrency,
+                           bool isBasisOnFxBaseCurrencyLeg) const {
+            const Real baseCcyLegNotional = 1.0;
+            Real quoteCcyLegNotional = baseCcyLegNotional * fxSpot;
+
+            Spread baseCcyLegBasis = isBasisOnFxBaseCurrencyLeg ? q.basis * basisPoint : 0.0;
+            Spread quoteCcyLegBasis = isBasisOnFxBaseCurrencyLeg ? 0.0 : q.basis * basisPoint;
+
+            std::vector<ext::shared_ptr<Swap> > legs;
+            ext::shared_ptr<Swap> baseCcyLeg = CrossCurrencyHelperFunctions::buildCrossCurrencyLeg(
+                start, Period(q.n, q.units), settlementDays, calendar, businessConvention,
+                endOfMonth, baseCcyIdx, VanillaSwap::Receiver, baseCcyLegNotional, baseCcyLegBasis);
+            legs.push_back(baseCcyLeg);
+
+            ext::shared_ptr<Swap> quoteCcyLeg = CrossCurrencyHelperFunctions::buildCrossCurrencyLeg(
+                start, Period(q.n, q.units), settlementDays, calendar, businessConvention,
+                endOfMonth, quoteCcyIdx, VanillaSwap::Payer, quoteCcyLegNotional, quoteCcyLegBasis);
+            legs.push_back(quoteCcyLeg);
+
+            return legs;
+        }
+
+        CommonVars() {
+            settlementDays = 2;
+            businessConvention = Following;
+            calendar = TARGET();
+            dayCount = Actual365Fixed();
+            endOfMonth = false;
+
+            basisPoint = 1.0e-4;
+            fxSpot = 1.25;
+
+            baseCcyIdx = ext::shared_ptr<IborIndex>(new Euribor3M(baseCcyIdxHandle));
+            quoteCcyIdx = ext::shared_ptr<IborIndex>(new USDLibor(3 * Months, quoteCcyIdxHandle));
+
+            /* Data source:
+               N. Moreni, A. Pallavicini (2015)
+               FX Modelling in Collateralized Markets: foreign measures, basis curves
+               and pricing formulae.
+
+               section 4.2.1, Table 2.
+            */
+            basisData.push_back(XccyTestDatum(1, Years, -14.5));
+            basisData.push_back(XccyTestDatum(18, Months, -18.5));
+            basisData.push_back(XccyTestDatum(2, Years, -20.5));
+            basisData.push_back(XccyTestDatum(3, Years, -23.75));
+            basisData.push_back(XccyTestDatum(4, Years, -25.5));
+            basisData.push_back(XccyTestDatum(5, Years, -26.5));
+            basisData.push_back(XccyTestDatum(7, Years, -26.75));
+            basisData.push_back(XccyTestDatum(10, Years, -26.25));
+            basisData.push_back(XccyTestDatum(15, Years, -24.75));
+            basisData.push_back(XccyTestDatum(20, Years, -23.25));
+            basisData.push_back(XccyTestDatum(30, Years, -20.50));
+
+            today = calendar.adjust(Date(6, September, 2013));
```

Comment:
Any particular reason to choose this evaluation date?

Reply:
This is the valuation date of the market data provided in the paper of Moreni & Pallavicini, which I reused in the test cases. I can change it to a more recent data, if needed, or just clarify it in the comment?

Reply:
No, that's ok, thanks